### PR TITLE
Makes having high body temperature more deadly in general

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
Current damage per Life():
image

This PR:
image

And while on fire:
image

:cl: Naksu
balance: Having a high body temperature now increases the damage you take gradually, whether you're on fire or not. Being on fire also always increases body temperature damage
/:cl:

### CODE REVIEW

**Comment by Cyprex** [comment]
Are you mapping temperature in Celsius or Kelvin? Also I feel like this should be testmerged once it's ready, as this might have large balance implications, fire and fire-based weaponry in general seems more dangerous with this.

**Comment by vuonojenmustaturska** [comment]
The charts are in celsius, and my idea is to use this to discover any warts preventing further scaling. I want to make having a high body temperature extremely deadly, but at the same time I don't want to have a period where eating a warm donk pocket husks you

**Comment by vuonojenmustaturska** [comment]
whether you wear one or not does not affect the damage you take from having a high body temperature

**Comment by Wilchenx** [comment]
this makes fire chems even more deadly

**Comment by Hate9** [comment]
Imagine this before sparks were nerfed.

**Comment by AdamElTablawy** [comment]
capsaicin-based botany plants are going to be nightmares, now, too.

**Comment by XDTM** [comment]
This makes sense, but balancing numbers based on advanced math is a pain for anyone that goes over that code afterwards

**Comment by Cyprex** [comment]
>burn_damage = log(2,(H.bodytemperature-BODYTEMP_NORMAL))-5
>advanced math
Hmmmmmmm

**Comment by XDTM** [comment]
Advanced in the sense you can't tell the range of possibile values quickly

I'm having similar problems with radiation while coding nanites, since i can't tell how many rads i can expect to get in average on a mob. Is 20 high? Low? What's the difference between standing at 3 or 5 tiles from a rad pulse?
It means i either need to manually calculate several potential scenarios or add a debug output in game to run them, which is time consuming.

**Comment by SgtxRootbeer** [comment]
How does this apply to lizards with their innate heat resistance?

**Comment by vuonojenmustaturska** [comment]
their heat resistance applies as normal

**Comment by blobbernaut** [comment]
Sure, but you should also look at how fast certain things raise body temp before you do this

**Comment by epochayur** [comment]
wait, does capsaicin actually raise your body temp? never heard of someone eating a habanero and then spontaneously combusting

**Comment by ninjanomnom** [comment]
Just throw the equation into a graph @XDTM, math is better than large amounts of code to do the same thing.

**Comment by Tlaltecuhtli** [comment]
being sprayed with clf3 once means you die even if you extinguish yourself in 5 seconds

**Comment by GuppyLaxx** [comment]
cant wait to die in even less seconds now as plasmaman

**Comment by Doppeltongue** [comment]
Seems like this would make fire viruses more lethal as well.

**Comment by Supermichael777** [comment]
Way to strong
Flamers are wipping out sec every damn round its just such an edgecase

**Comment by vuonojenmustaturska** [comment]
just had a follow-up idea i want to add to this

**Comment by vuonojenmustaturska** [comment]
damage for not being on fire unchanged, old revision being on fire is equivalent to 15 fire stacks new revision

**Comment by bigfatbananacyclops** [comment]
i went into crit in 2 seconds and died with this fire

**Comment by Bod9001** [comment]
I felt like before this patch the fire did a minimal amount of damage the majority of the damage was from the character being too hot, Maybe balance it to the character actually being on fire does the damage and the heat damaged does organ damage instead.

**Comment by XDTM** [comment]
We don't have an organ damage system yet, although we're close

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -1517,19 +1517,20 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
-			if(400 to 460)
-				H.throw_alert("temp", /obj/screen/alert/hot, 2)
-				burn_damage = HEAT_DAMAGE_LEVEL_2
-			else
-				H.throw_alert("temp", /obj/screen/alert/hot, 3)
-				if(H.on_fire)
-					burn_damage = HEAT_DAMAGE_LEVEL_3
+		var/firemodifier = H.fire_stacks / 5
+		if (H.on_fire)
+			burn_damage = log(2-firemodifier,(H.bodytemperature-BODYTEMP_NORMAL))-5
```
Comment (AnturK) [inline code/modules/mob/living/carbon/human/species.dm]: Needs clamp to 0 in this block too.

**Review by AnturK — DISMISSED** [review]

File: code/modules/mob/living/carbon/human/species.dm (hunk 1 above)
Comment (vuonojenmustaturska) [reply code/modules/mob/living/carbon/human/species.dm]: ree

**Comment by blobbernaut** [comment]
This needs a fucking revert.

**Comment by optimumtact** [comment]
its' test merged you low kino poster

**Comment by blobbernaut** [comment]
yeah and the test's pretty clear that it sucks

**Comment by blobbernaut** [comment]
This is shit and bad

**Pull request merged**